src/Protocol/Protocol_1_8.cpp

```diff
@@ -2969,6 +2980,65 @@ void cProtocol_1_8_0::HandleVanillaPluginMessage(cByteBuffer & a_ByteBuffer, con
 		m_Client->HandleNPCTrade(SlotNum);
 		return;
 	}
+	else if ((a_Channel == "MC|BSign") || (a_Channel == "MC|BEdit"))
+	{
+		HANDLE_READ(a_ByteBuffer, ReadBEInt16, Int16, ItemID);
+		// In 1.8 the item id is 387 for a signed book, instead of 386
+		if ((ItemID != E_ITEM_WRITTEN_BOOK) && (ItemID != E_ITEM_BOOK_AND_QUILL))
+		{
+			// Item is not a writeable book
+			return;
+		}
+
+		cPlayer & Player = *m_Client->GetPlayer();
+		if (Player.GetEquippedItem().m_ItemType != E_ITEM_BOOK_AND_QUILL)
+		{
+			// Equipped item is not a writeable book
+			return;
+		}
+
+		// Skip item count (1 byte) and item damage (2 bytes)
+		a_ByteBuffer.SkipRead(3);
+
+		// Read nbt content
+		AString BookData;
+		a_ByteBuffer.ReadString(BookData, a_ByteBuffer.GetReadableSpace() - 1);
+		cParsedNBT NBT(BookData.c_str(), BookData.size());
+
+		cItem BookItem;
+		bool IsSigned = true;
+		if (a_Channel == "MC|BSign")
+		{
+			BookItem = cItem(E_ITEM_WRITTEN_BOOK);
+			// Add the text to a json string
+			cBookContent::ParseFromNBT(0, BookItem.m_BookContent, NBT, true);
+		}
+		else
+		{
+			IsSigned = false;
+			BookItem = cItem(E_ITEM_BOOK_AND_QUILL);
+			cBookContent::ParseFromNBT(0, BookItem.m_BookContent, NBT);
+		}
+		BookItem.m_BookContent.SetIsSigned(IsSigned);
+
+		// The equipped item contains the old book content
+		if (cRoot::Get()->GetPluginManager()->CallHookPlayerEditingBook(Player, Player.GetEquippedItem().m_BookContent, BookItem.m_BookContent, IsSigned))
+		{
+			// Plugin denied the player to edit the book
```

Comment:
Again, plugin denied editing but we do it anyway?

Reply:
Fixed, thanks.